src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/Network/LinuxNetworkUtilizationParser.cs

```diff
@@ -0,0 +1,163 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.ObjectPool;
+#if !NET8_0_OR_GREATER
+using Microsoft.Shared.StringSplit;
+#endif
+using Microsoft.Shared.Diagnostics;
+using Microsoft.Shared.Pools;
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Linux.Network;
+
+internal sealed class LinuxNetworkUtilizationParser
+{
+    private static readonly ObjectPool<BufferWriter<char>> _sharedBufferWriterPool = BufferWriterPool.CreateBufferWriterPool<char>();
+
+    /// <remarks>
+    /// File that provide information about currently active TCP_IPv4 connections.
+    /// </remarks>
+    private static readonly FileInfo _tcp = new("/proc/net/tcp");
+
+    /// <remarks>
+    /// File that provide information about currently active TCP_IPv6 connections.
+    /// </remarks>
+    private static readonly FileInfo _tcp6 = new("/proc/net/tcp6");
+
+    private readonly IFileSystem _fileSystem;
+
+    /// <remarks>
+    /// Reads the contents of a file located at <see cref="_tcp"/> and parses it to extract information about the TCP/IP state info of the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv4StateInfo() => GetTcpStateInfo(_tcp);
+
+    /// <remarks>
+    /// Reads the contents of a file located at <see cref="_tcp6"/> and parses it to extract information about the TCP/IP state info of the system.
+    /// </remarks>
+    public TcpStateInfo GetTcpIPv6StateInfo() => GetTcpStateInfo(_tcp6);
+
+    public LinuxNetworkUtilizationParser(IFileSystem fileSystem)
+    {
+        _fileSystem = fileSystem;
+    }
+
+    /// <remarks>
+    /// Parses the contents of the <paramref name="buffer"/> and updates the <paramref name="tcpStateInfo"/> with the parsed information.
+    /// For the data format expected in the <paramref name="buffer"/>, refer <see href="https://www.kernel.org/doc/Documentation/networking/proc_net_tcp.txt">proc net tcp</see>.
+    /// </remarks>
+    private static void UpdateTcpStateInfo(ReadOnlySpan<char> buffer, TcpStateInfo tcpStateInfo)
+    {
+        const int Base16 = 16;
+
+        // The buffer contains one line from /proc/net/tcp(6) file, e.g.:
+        //   0: 030011AC:8AF2 C1B17822:01BB 01 00000000:00000000 02:000000D1 00000000   472        0 2481276 2 00000000c62511cb 28 4 26 10 -1
+        // The line may contain leading spaces, so we have to trim those.
+        // tcpConnectionState is in the 4th column - i.e., "01".
+        ReadOnlySpan<char> line = buffer.TrimStart();
+
+#if NET8_0_OR_GREATER
+        const int Target = 5;
+        Span<Range> range = stackalloc Range[Target];
+
+        // on .NET 8+, if capacity of destination range array is less than number of ranges found by ReadOnlySpan<T>.Split(),
+        // the last range in the array will get all the remaining elements of the ReadOnlySpan.
+        // therefore we request 5 ranges instead of 4, and then range[Target - 2] will have the range we need without the remaining elements.
```

Comment:
```suggestion
        // In .NET 8+, if capacity of destination range array is less than number of ranges found by ReadOnlySpan<T>.Split(),
        // the last range in the array will get all the remaining elements of the ReadOnlySpan.
        // Therefore, we request 5 ranges instead of 4, and then range[Target - 2] will have the range we need without the remaining elements.
```

Reply:
Done